File: data/cpp/tests/test_sortedcontainers_SortedList_bisect_left.cpp
#include <gtest/gtest.h>
#include "../src/function_sortedcontainers_SortedList_bisect_left.cpp"

class TestSortedListBisectLeftMethod : public ::testing::Test {
protected:
    HanSeok handler;
};

TEST_F(TestSortedListBisectLeftMethod, test_value_in_middle) {
    int position = handler.find_insert_position(4);
    EXPECT_EQ(position, 2);
}

TEST_F(TestSortedListBisectLeftMethod, test_value_at_start) {
    int position = handler.find_insert_position(0);
    EXPECT_EQ(position, 0);
}

TEST_F(TestSortedListBisectLeftMethod, test_value_at_end) {
    int position = handler.find_insert_position(10);
    EXPECT_EQ(position, 4);
}

TEST_F(TestSortedListBisectLeftMethod, test_existing_value) {
    int position = handler.find_insert_position(3);
    EXPECT_EQ(position, 1);
}

TEST_F(TestSortedListBisectLeftMethod, test_insert_into_empty_list) {
    int position = handler.find_insert_position(5, "5");
    EXPECT_EQ(position, 0);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/java/tests/test_sortedcontainers_SortedList_bisect_left.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;

public class test_sortedcontainers_SortedList_bisect_left {

    @Test
    public void test_value_in_middle() {
        function_sortedcontainers_SortedList_bisect_left instance = new function_sortedcontainers_SortedList_bisect_left();
        int position = instance.find_insert_position(4, null);
        assertEquals(2, position);
    }

    @Test
    public void test_value_at_start() {
        function_sortedcontainers_SortedList_bisect_left instance = new function_sortedcontainers_SortedList_bisect_left();
        int position = instance.find_insert_position(0, null);
        assertEquals(0, position);
    }

    @Test
    public void test_value_at_end() {
        function_sortedcontainers_SortedList_bisect_left instance = new function_sortedcontainers_SortedList_bisect_left();
        int position = instance.find_insert_position(10, null);
        assertEquals(4, position);
    }

    @Test
    public void test_existing_value() {
        function_sortedcontainers_SortedList_bisect_left instance = new function_sortedcontainers_SortedList_bisect_left();
        int position = instance.find_insert_position(3, null);
        assertEquals(1, position);
    }

    @Test
    public void test_insert_into_empty_list() {
        function_sortedcontainers_SortedList_bisect_left instance = new function_sortedcontainers_SortedList_bisect_left();
        int position = instance.find_insert_position(5, "5");
        assertEquals(0, position);
    }
}